clang/lib/CodeGen/Targets/AMDGPU.cpp

```diff
@@ -103,19 +104,27 @@ void AMDGPUABIInfo::computeInfo(CGFunctionInfo &FI) const {
   if (!getCXXABI().classifyReturnType(FI))
     FI.getReturnInfo() = classifyReturnType(FI.getReturnType());
 
+  unsigned ArgumentIndex = 0;
+  const unsigned numFixedArguments = FI.getNumRequiredArgs();
```

Comment:
Can you split the clang AMDGPU ABI changes into a separate PR? The tests for this are also missing

Reply:
Yes, I think I can do - the ABI change only affects variadic functions, which currently fatal_error anyway - but I think the C to IR tests will succeed as long as nothing calls va_arg and it stops before codegen.

Reply:
Added the test to this PR and also split out #94083. Can land that subpatch first and rebase this for a reduction in complexity.